eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubmittedVariantDeclusterProcessorTest.java

```diff
@@ -0,0 +1,192 @@
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static uk.ac.ebi.eva.accession.core.ISubmittedVariant.DEFAULT_ASSEMBLY_MATCH;
+import static uk.ac.ebi.eva.accession.core.ISubmittedVariant.DEFAULT_SUPPORTED_BY_EVIDENCE;
+import static uk.ac.ebi.eva.accession.core.ISubmittedVariant.DEFAULT_VALIDATED;
+import static uk.ac.ebi.eva.accession.dbsnp.processors.SubmittedVariantDeclusterProcessor.DECLUSTERED;
+import static uk.ac.ebi.eva.accession.dbsnp.processors.SubmittedVariantDeclusterProcessor.DECLUSTERED_ALLELES_MISMATCH;
+import static uk.ac.ebi.eva.accession.dbsnp.processors.SubmittedVariantDeclusterProcessor.DECLUSTERED_TYPE_MISMATCH;
+
+public class SubmittedVariantDeclusterProcessorTest {
+
+    private static final String ASSEMBLY = "AnyAssembly-1.0";
+
+    private static final String PROJECT_ACCESSION = "PROJECT";
+
+    private static final String CONTIG = "CONTIG";
+
+    private static final int TAXONOMY = 3880;
+
+    private static final int START = 100;
+
+    private static final Long CLUSTERED_VARIANT_ACCESSION = 12L;
+
+    private static final Long SUBMITTED_VARIANT_ACCESSION = 15L;
+
+    private static SubmittedVariantDeclusterProcessor processor;
+
+    private static Function<ISubmittedVariant, String> hashingFunctionSubmitted;
+
+    private static Function<IClusteredVariant, String> hashingFunctionClustered;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        processor = new SubmittedVariantDeclusterProcessor();
+        hashingFunctionSubmitted = new DbsnpSubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
+        hashingFunctionClustered = new DbsnpClusteredVariantSummaryFunction().andThen(new SHA1HashingFunction());
+    }
+
+    @Test
+    public void processSubmittedVariantAllelesMismatch() throws Exception {
+        List<DbsnpSubmittedVariantEntity> submittedVariantEntities = createSubmittedVariantEntities("A", "C", false);
+        DbsnpVariantsWrapper wrapper = buildSimpleWrapper(submittedVariantEntities, DbsnpVariantType.SNV,
+                                                          VariantType.SNV);
+        processor.process(wrapper);
+
+        //Check Decluster
+        assertEquals(1, wrapper.getSubmittedVariants().size());
+        assertNull(wrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+
+        //Check Operations
+        assertOperation(wrapper, getReason(Collections.singletonList(DECLUSTERED_ALLELES_MISMATCH)),
+                        wrapper.getSubmittedVariants().get(0).getAccession(), 0);
+    }
+
+    @Test
+    public void processSubmittedVariantTypeMismatch() throws Exception {
```

Comment:
The input for this method is the same as in `processSubmittedVariantAllelesAndTypeMismatch`. Modify this one so it only checks the case of type mismatch.

Reply:
The difference in the input is the `allelesMatch` flag. In this case the value is `true` so it will not decluster for alleles mismatch, only in case of type mismatch. 

The result of this test is 2 variants declustered out of 3, while the `processSubmittedVariantAllelesAndTypeMismatch` has all the variants declustered.

Reply:
I think it is very counter-intuitive to build an object with alleles that mismatch, but the flag is set to express the opposite...

Reply:
The input in this processor is the wrapper and it doesn't have a variant with the same reference and alternate, it only has the 'allelesMatch` flag on every variant. If we add the reference in the alternates list it will create a variant that shouldn't be in the wrapper.  

`SubSnpNoHgvsToDbsnpVariantsWrapperProcessor` is called before the decluster and it calculates `allelesMatch` flag and create one variant for each alternate allele except when the alternate is the same reference.